### DISCUSSION TITLE
how to judge the workflow is over

### DISCUSSION THREAD

**Discussion by simonli866** [StackStorm/st2#5813] [Workflows]
I read the source code and saw a lot of status updates. But how does the system know if workflow or action is complete, and where is the code for this part?

**Comment by ip-sf** [comment]
Assuming you are using orquesta, The workflow completes when the end of the graph is reached. It's something that can be inferred logically/implicitly and to my knowledge, is. Similar in behavior is entry point. You can specify an entry point, but otherwise the execution starts at the 'top' of the graph. If I recall correctly, if you have what would be 2 or more 'top' then you would have to explicitly specify the entrypoint in the workflow's metadata.

HTH
